npm_and_yarn/lib/dependabot/npm_and_yarn.rb

```diff
@@ -25,3 +25,101 @@
     groups.include?("dependencies")
   end
 )
+
+module Dependabot
+  module NpmAndYarn
+    YARN_CODE_REGEX = /(YN\d{4})/
+    YARN_ERROR_CODES = T.let({
+      "YN0000" => { message: "Unnamed error", error_class: Dependabot::DependabotError },
+      "YN0001" => { message: "Exception error", error_class: Dependabot::DependabotError },
+      "YN0002" => { message: "Missing peer dependency", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0003" => { message: "Cyclic dependencies", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0004" => { message: "Disabled build scripts", error_class: Dependabot::DependabotError },
+      "YN0005" => { message: "Build disabled", error_class: Dependabot::DependabotError },
+      "YN0006" => { message: "Soft link build", error_class: Dependabot::DependabotError },
+      "YN0007" => { message: "Must build", error_class: Dependabot::DependabotError },
+      "YN0008" => { message: "Must rebuild", error_class: Dependabot::DependabotError },
+      "YN0009" => { message: "Build failed", error_class: Dependabot::DependabotError },
+      "YN0010" => { message: "Resolver not found", error_class: Dependabot::DependabotError },
+      "YN0011" => { message: "Fetcher not found", error_class: Dependabot::DependabotError },
+      "YN0012" => { message: "Linker not found", error_class: Dependabot::DependabotError },
+      "YN0013" => { message: "Fetch not cached", error_class: Dependabot::DependabotError },
+      "YN0014" => { message: "Yarn import failed", error_class: Dependabot::DependabotError },
+      "YN0015" => { message: "Remote invalid", error_class: Dependabot::DependabotError },
+      "YN0016" => { message: "Remote not found", error_class: Dependabot::GitDependenciesNotReachable },
+      "YN0017" => { message: "Resolution pack error", error_class: Dependabot::DependabotError },
+      "YN0018" => { message: "Cache checksum mismatch", error_class: Dependabot::DependabotError },
+      "YN0019" => { message: "Unused cache entry", error_class: Dependabot::DependabotError },
+      "YN0020" => { message: "Missing lockfile entry", error_class: Dependabot::DependabotError },
+      "YN0021" => { message: "Workspace not found", error_class: Dependabot::DependabotError },
+      "YN0022" => { message: "Too many matching workspaces", error_class: Dependabot::DependabotError },
+      "YN0023" => { message: "Constraints missing dependency", error_class: Dependabot::DependabotError },
```

Comment:
I think it'll be useful to map these known yarn errors to the appropriate type of Dependabot errors so that they are actionable by our users.

Reply:
I will implement them after testing existing ones. I did't create new error, just used existing ones. I will check if we have already existing error matching any code, if not then I need to create them.

Reply:
Unnecessary error codes are removed and some new Error classes are added to specify each yarn error code.